### DISCUSSION TITLE
OctoText Doesn't Appear in Plugins Menu

### DISCUSSION THREAD

**Discussion by jcjenk** [berrystephenw/OctoText#5] [Q&A]
I have installed, activated, and rebooted OctoPrint but the OctoText plugin does not appear in the menu?

![OctoText](https://user-images.githubusercontent.com/31355763/113060598-b9364c80-9165-11eb-8122-3085eaf8421e.png)

My system information:
browser.user_agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.90 Safari/537.36
connectivity.connection_check: 8.8.8.8:53
connectivity.connection_ok: true
connectivity.enabled: true
connectivity.online: true
connectivity.resolution_check: octoprint.org
connectivity.resolution_ok: true
env.hardware.cores: 4
env.hardware.freq: 1200
env.hardware.ram: 918192128
env.os.bits: 32
env.os.id: linux
env.os.platform: linux2
env.plugins.pi_support.model: Raspberry Pi 3 Model B Rev 1.2
env.plugins.pi_support.octopi_version: 0.15.0
env.plugins.pi_support.throttle_state: 0x0
env.python.pip: 10.0.1
env.python.version: 2.7.13
env.python.virtualenv: true
octoprint.safe_mode: false
octoprint.version: 1.5.3
printer.firmware: Marlin 2.0.3 (GitHub)

**Comment by berrystephenw** [comment]
Hi, thanks for trying it, but I think the problem is that your version of Python is too old for the code. It requires python 3 to install correctly. I’m going to see if I can find out why it didn’t flag that before you installed!

Sorry about that! There is a way to upgrade but it might affect your other plugins. Let me know if you want to do that and I’ll find the instructions for you.

Steve

Stephen Berry
Hardware Engineering Manager
GSSI
40 Simon St
Nashua, NH 03060
________________________________
From: jcjenk ***@***.***>
Sent: Tuesday, March 30, 2021 5:43:22 PM
To: berrystephenw/OctoText ***@***.***>
Cc: Subscribed ***@***.***>
Subject: [berrystephenw/OctoText] OctoText Doesn't Appear in Plugins Menu (#5)


I have installed, activated, and rebooted OctoPrint but the OctoText plugin does not appear in the menu?

[OctoText]<https://user-images.githubusercontent.com/31355763/113060598-b9364c80-9165-11eb-8122-3085eaf8421e.png>

My system information:
browser.user_agent: Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/89.0.4389.90 Safari/537.36
connectivity.connection_check: 8.8.8.8:53
connectivity.connection_ok: true
connectivity.enabled: true
connectivity.online: true
connectivity.resolution_check: octoprint.org
connectivity.resolution_ok: true
env.hardware.cores: 4
env.hardware.freq: 1200
env.hardware.ram: 918192128
env.os.bits: 32
env.os.id: linux
env.os.platform: linux2
env.plugins.pi_support.model: Raspberry Pi 3 Model B Rev 1.2
env.plugins.pi_support.octopi_version: 0.15.0
env.plugins.pi_support.throttle_state: 0x0
env.python.pip: 10.0.1
env.python.version: 2.7.13
env.python.virtualenv: true
octoprint.safe_mode: false
octoprint.version: 1.5.3
printer.firmware: Marlin 2.0.3 (GitHub)

**Reply by jcjenk** [reply]
I might try to upgrade python.  I'll double-check the impact on my other plugins first.  Thank You.

**Reply by thingmaker33** [reply]
I'm having the same problem.  Seems all my other plug-ins show >= 2.7.
Could you let me know how your upgrade went?  Thank you.

**Reply by thingmaker33** [reply]
UPDATE for me:  I upgraded to Octopi .18 and OctoPrint 1.5.3 and everything works.  Thank you for a great plug in.  Texting works!

**Reply by berrystephenw** [reply]
Glad to hear it!

Sorry about the requirement to upgrade to Python 3, the plugin just doesn't install under 2.7 because of the email package - and that is essential to the operation.

**Reply by jcjenk** [reply]
I updated to OctoPi 0.18.0 then OctoPrint 1.5.3 which left me with Python 3.7.3 and OctoText now works great.  Thank you.

I used this to "backup" by current plugin settings https://community.octoprint.org/t/how-do-i-backup-my-octoprint-settings-on-octopi/1489.

It saved some of them but not all of them.  It didn't save my GCODE settings for the bed level visualizer.
